Gems/ROS2/Code/Source/Frame/ROS2Transform.cpp

```diff
@@ -19,6 +20,20 @@ namespace ROS2
         , m_childFrame(AZStd::move(childFrame))
         , m_isDynamic(isDynamic)
     {
+        auto splitName = ROS2Names::SplitIntoNamespaceAndName(m_childFrame);
+        if (splitName.first.empty())
+        {   // No namespace in child frame, no need for a publisher.
+            return;
+        }
+
+        auto ros2Node = ROS2Interface::Get()->GetNode();
+        rclcpp::QoS qos = tf2_ros::StaticBroadcasterQoS();
+        if (m_isDynamic)
+        {
+            qos = tf2_ros::DynamicBroadcasterQoS();
+        }
+        AZStd::string fullTopic = ROS2Names::GetNamespacedName(splitName.first, m_isDynamic ? "tf" : "tf_static");
+        m_transformPublisher = ros2Node->create_publisher<geometry_msgs::msg::TransformStamped>(fullTopic.data(), qos);
```

Comment:
In my opinion not every namespace should have separate TF tree.
Ex. robotic arm with many namespaced joints does not need seperate tf topic for each joint.
I believe this should be an opt in feature.

Reply:
I do not see a satisfying solution in terms of UX at the moment, any suggestions?